### PR TITLE
feat: add nightly release

### PR DESCRIPTION
**Pull request by WenheLI**
This PR adds a nightly release support

### CODE REVIEW

File: docs/meta/PROJECT_GUIDE.md [hunk 1]
```diff
@@ -74,3 +74,21 @@ To help Pipcook discover all the plugins, the project provides some rules to let
 Community plugins can also be submitted as built-in plugins through pull requests, but this requires nomination by a core collaborator and the approvals of at least 2 collaborators.
 
 [semver]: https://semver.org/
+
+### Nightly Release
+
+The Nightly release will be built and published to npm every midnight if there is any commit in the past day.
```
Comment (yorkie) [inline docs/meta/PROJECT_GUIDE.md:80]:
```suggestion
The nightly release will be built and published to npm every midnight if there is any commit in the past day.
```

File: docs/meta/PROJECT_GUIDE.md [hunk 2]
```diff
@@ -74,3 +74,21 @@ To help Pipcook discover all the plugins, the project provides some rules to let
...
+The aim to have such a release is to provide a quick fix to existig bugs and access to new/experimental features that are not suitable to deploy under the production environment.
```
Comment (yorkie) [inline docs/meta/PROJECT_GUIDE.md:81]:
```suggestion
The aim to have such a release is to provide a quick fix to existing bugs and access to new/experimental features that are not suitable to deploy under the production environment.
```

File: docs/meta/PROJECT_GUIDE.md [hunk 3]
```diff
@@ -74,3 +74,21 @@ To help Pipcook discover all the plugins, the project provides some rules to let
...
+
+#### Usage
+
+You can directly use the following command:
+```bash
+npm install @pipcook/pipcook-cli@nightly
```
Comment (yorkie) [inline docs/meta/PROJECT_GUIDE.md:87]:
```suggestion
$ npm install @pipcook/pipcook-cli@nightly
```

Comment (yorkie) [reply docs/meta/PROJECT_GUIDE.md:87]: Oh, shall we use beta as the npm tag? See https://github.com/alibaba/pipcook/blob/master/packages/cli/src/actions/init.ts#L28.

Or we need to change it `pipcook init --nightly`? /cc @FeelyChau @rickycao-qy

File: docs/meta/PROJECT_GUIDE.md [hunk 4]
```diff
@@ -74,3 +74,21 @@ To help Pipcook discover all the plugins, the project provides some rules to let
...
+```
+
+Or
+
+```bash
+npm install --tag nightly @pipcook/pipcook-cli
```
Comment (yorkie) [inline docs/meta/PROJECT_GUIDE.md:93]:
```suggestion
$ npm install --tag nightly @pipcook/pipcook-cli
```

File: docs/meta/PROJECT_GUIDE.md (hunk 3 above)
Comment (WenheLI) [reply docs/meta/PROJECT_GUIDE.md:87]: I see. I will use `beta` as the nightly release tag

File: .github/workflows/nightly.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        if: env.UPDATED == 'true'
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.npm_token }}" > ~/.npmrc
+          git config --global user.name 'pipcook'
+          git config --global user.email 'queyue.crk@alibaba-inc.com'
+          npm run nightly-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
+          NPM_AUTH_TOKEN: ${{ secrets.npm_token }}
```
Comment (rickycao-qy) [inline .github/workflows/nightly.yml:61]: EOF

File: docs/meta/PROJECT_GUIDE.md (hunk 3 above)
Comment (yorkie) [reply docs/meta/PROJECT_GUIDE.md]: Remember updating documentation, either :p

File: .github/workflows/nightly.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
+name: Nightly Build Packages
```
Comment (yorkie) [inline .github/workflows/nightly.yml:1]:
```suggestion
name: Beta Build Packages
```

Comment (yorkie) [reply .github/workflows/nightly.yml:1]: Should we change the workflow to beta.yml?

File: .github/workflows/nightly.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branches: nightly
+
+      - name: Create Nightly Branch
+        uses: peterjgrainger/action-create-branch@v1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: 'nightly'
```
Comment (yorkie) [inline .github/workflows/nightly.yml:22]: Ditto

File: .github/workflows/nightly.yml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Checkout nightly
+        uses: actions/checkout@v2
+        with:
+          ref: nightly
```
Comment (yorkie) [inline .github/workflows/nightly.yml:27]: Ditto

File: docs/meta/PROJECT_GUIDE.md [hunk 9]
```diff
@@ -74,3 +74,21 @@ To help Pipcook discover all the plugins, the project provides some rules to let
...
+### Beta Release
+
+The beta release will be built and published to npm every midnight if there is any commit in the past day.
+The aim of having such a release is to provide a quick fix to existing bugs and access to new/experimental features that are not suitable to deploy under the production environment.
+
+#### Usage
+
+You can directly use the following command:
+```bash
+$ npm install @pipcook/pipcook-cli@beta
```
Comment (yorkie) [inline docs/meta/PROJECT_GUIDE.md:87]:
```suggestion
$ npm install @pipcook/pipcook-cli@beta -g
```

File: docs/meta/PROJECT_GUIDE.md [hunk 10]
```diff
@@ -74,3 +74,21 @@ To help Pipcook discover all the plugins, the project provides some rules to let
...
+```
+
+Or
+
+```bash
+$ npm install --tag beta @pipcook/pipcook-cli
```
Comment (yorkie) [inline docs/meta/PROJECT_GUIDE.md:93]:
```suggestion
$ npm install --tag beta @pipcook/pipcook-cli -g
```

**Comment by rickycao-qy** [comment]
LGTM

**Pull request merged**